test/lib/api_translator_test.rb

```diff
@@ -58,11 +63,53 @@ def test_translate_without_api_compression_sends_json
                          'product' => 'WOVN.rb',
                          'version' => VERSION,
                          'body' => 'foo',
+                         'user_agent' => 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.81 Safari/537.36',
                          'custom_lang_aliases' => { 'ja' => 'Japanese' }.to_json
                        }.to_json,
                        times: 1
     end
 
+    def test_translate_is_search_engine_bot
+      settings = {
+        'project_token' => '123456',
+        'custom_lang_aliases' => { 'ja' => 'Japanese' },
+        'default_lang' => 'en',
+        'url_pattern' => 'subdomain',
+        'url_pattern_reg' => '^(?<lang>[^.]+).',
+        'lang_param_name' => 'lang',
+        'compress_api_requests' => false
+      }
+      store = Wovnrb::Store.instance
+      store.update_settings(settings)
+      env = Wovnrb.get_env('url' => 'http://fr.wovn.io/test')
+      env['HTTP_USER_AGENT'] = 'Googlebot/'
+      headers = Wovnrb::Headers.new(
+        env,
+        Wovnrb.get_settings(settings),
+        UrlLanguageSwitcher.new(store)
+      )
+      html_body = 'foo'
+
+      time_proc = -> { return 25_000_000 }
+      api_translator = ApiTranslator.new(store, headers, REQUEST_UUID, time_proc)
+      stub_request(:post, %r{http://wovn\.global\.ssl\.fastly\.net/v0/translation\?cache_key=.*timestamp=1970-10-17T08:20:00%2B00:00.*})
+        .to_return(status: 200, body: { 'body' => 'translated_body' }.to_json)
+
+      api_translator.translate(html_body)
+
+      time_proc = -> { return 25_000_300 }
+      api_translator = ApiTranslator.new(store, headers, REQUEST_UUID, time_proc)
```

Comment:
I don't think we need this kind of design in Ruby? It is common to use the Timecop gem

Reply:
No we don't. But I want to keep the implementation across different languages as close as possible.